### PR TITLE
fix reconnecting while set switchbutton to unactive

### PR DESCRIPTION
**Pull request by anzyelay**
to resolve  "Network settings panel keeps reconnecting wired network #326 "

### CODE REVIEW

File: src/Widgets/Page.vala [hunk 1]
```diff
@@ -70,11 +71,44 @@ namespace Network.Widgets {
...
+                    status_switch.active = true;
+                    switch_updating = false;
+                    break;
+            }
+            //  warning ("update switch activated to ==============%s", Utils.state_to_string (device.state));
         }
 
         protected virtual void control_switch_activated () {
+            if(switch_updating) 
+                return;
```
Comment (jeremypw) [inline src/Widgets/Page.vala:109]: Elementary codestyle:
```suggestion
            if (switch_updating) {
                return;
            }
```

**Review by jeremypw — CHANGES_REQUESTED** [review]
This successfully stabilises the Ethernet reconnection under circumstances described in
#326 and #312 - thanks for the fix!

Just a comment about Elementary code style.  Also although the switch statement seems to cover all possible cases, it is generally good practice to include a fallback `default:` case, even if it is only `assert_not_reached ()`.

Also please remove the commented out code.

File: src/Widgets/Page.vala (hunk 1 above)
Comment (anzyelay) [reply src/Widgets/Page.vala:109]: fix elementary code style

**Review by jeremypw — APPROVED** [review]

**Comment by Mepth** [comment]
Like!

**Pull request merged**